src/state/get-droppable-over.js

```diff
@@ -29,13 +66,17 @@ export default ({ target, droppables }: Args): ?DroppableId => {
         return false;
       }
 
-      // Not checking to see if visible in viewport
-      // as the target might be off screen if dragging a large draggable
-      // Not adjusting target for droppable scroll as we are just checking
-      // if it is over the droppable - not its internal impact
-      return isPositionInFrame(active)(target);
+      return isCandidate({ active, target, client: draggable.client });
     },
   );
 
-  return maybe ? maybe.descriptor.id : null;
+  const origin = droppables[draggable.descriptor.droppableId];
+
+  const preferences = active.sort(
+    (prev, next) =>
+      distance(origin.client.borderBox.center, next.client.borderBox.center) -
+      distance(origin.client.borderBox.center, prev.client.borderBox.center),
+  );
```

Comment:
Preferences are based on distance from the originating `droppable` center point. This might have issues for elements that are dragged from the top of one `droppable` and down to the bottom of another really long `droppable`...